core/src/components/modal/modal.tsx

```diff
@@ -426,19 +427,9 @@ export class Modal implements ComponentInterface, OverlayInterface {
       return;
     }
 
-    const { presentingElement, el } = this;
+    const unlock = await this.lockController.lock();
```

Comment:
For modal and popover we'll want to acquire the lock before we do the `this.presented` check otherwise the modal will just return early if you try to present, dismiss, then present quickly.